js/core/form/editors/AvatarEditorView.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * Developer: Oleg Verevkin
+ * Date: 02/19/2016
+ * Copyright: 2009-2016 Comindware®
+ *       All Rights Reserved
+ * Published under the MIT license
+ */
+
+'use strict';
+
+import {helpers} from '../../utils/utilsApi';
+import BaseItemEditorView from './base/BaseItemEditorView';
+import template from './templates/avatarEditor.hbs';
+
+const defaultOptions = {
+    autoUpload: false,
+    refreshPreviewAfterUpload: false
+};
+
+Backbone.Form.editors.Avatar = BaseItemEditorView.extend({
+    className: 'dev-avatar-editor',
+
+    attributes: {
+        tabindex: 0
+    },
+    
+    focusElement: null,
+    
+    template: template,
+    
+    ui: {
+        initials: '.js-initials',
+        tooltip: '.js-tooltip'
+    },
+    
+    events: {
+        click: '__attach'
+    },
+    
+    initialize(options) {
+        if (options.schema) {
+            _.extend(this.options, defaultOptions, _.pick(options.schema, _.keys(defaultOptions)));
+        } else {
+            _.extend(this.options, defaultOptions, _.pick(options || {}, _.keys(defaultOptions)));
+        }
+        
+        helpers.ensureOption(options, 'controller');
+        this.controller = this.getOption('controller');        
+        
+        this.__previewURL = null;
+        this.__initFileInput();
+    },
+    
+    onRender() {
+        if (this.getValue()) {
+            this.__preview(this.controller.getImage(this.getValue()));
+        } else if (this.getOption('fullName')) {
+            this.ui.initials.append(this.__getInitials(this.getOption('fullName')));
+            this.ui.initials.show();
+        } else {
+            this.__preview(this.controller.getImage());
+        }
+        
+        this.ui.tooltip.hide();
+        
+        this.$el.hover(
+            () => {
+                this.ui.tooltip.show();
+            },
+            () => {
+                this.ui.tooltip.hide();
+            }
+        );
+    },
+    
+    onBeforeDestroy() {
+        if (this.__previewURL) {
+            URL.revokeObjectURL(this.__previewURL);
+        }
+    },
+    
+    upload() {
+        if (this.fileInput.files[0]) {
```

Comment:
It's better to return Promise.resolve() otherwise so that the code `editor.upload().then(function () {})` always works.